File: BackEnd/EntreBicis/src/main/java/MarioOlaya/Projecte4/EntreBicis/entity/StateType.java
/**
 * Representa els diferents estats possibles per a un objecte en el sistema.
 * 
 * <ul>
 *   <li><b>DISPONIBLE</b>: L'objecte està disponible per ser utilitzat.</li>
 *   <li><b>RESERVADA</b>: L'objecte ha estat reservat per un usuari.</li>
 *   <li><b>ASSIGNADA</b>: L'objecte ha estat assignat a un usuari específic.</li>
 *   <li><b>RECOLLIDA</b>: L'objecte ha estat recollit per l'usuari.</li>
 * </ul>
 */
package MarioOlaya.Projecte4.EntreBicis.entity;

public enum StateType {
    DISPONIBLE, RESERVADA, ASSIGNADA, RECOLLIDA
}

File: FrontEnd/app/src/main/java/cat/copernic/entrebicis/Core/Model/StateType.kt
package cat.copernic.entrebicis.Core.Model

/**
 * Enumeració que representa els diferents estats que pot tenir una recompensa.
 *
 * - **DISPONIBLE**: La recompensa està disponible per ser reservada per un usuari.
 * - **RESERVADA**: La recompensa ha estat reservada per un usuari però encara no ha estat assignada.
 * - **ASSIGNADA**: La recompensa ha estat assignada a un usuari i està preparada per ser recollida.
 * - **RECOLLIDA**: La recompensa ja ha estat recollida pel seu destinatari.
 */
enum class StateType {
    DISPONIBLE,
    RESERVADA,
    ASSIGNADA,
    RECOLLIDA
}
